ui/store/actions.ts

```diff
@@ -2457,60 +2456,88 @@ export function setProviderType(
   };
 }
 
-export function updateAndSetCustomRpc(
-  newRpcUrl: string,
-  chainId: string,
-  ticker: EtherDenomination,
-  nickname: string,
-  rpcPrefs: RPCDefinition['rpcPrefs'],
+export function upsertNetworkConfiguration(
+  {
+    rpcUrl,
+    chainId,
+    nickname,
+    rpcPrefs,
+    ticker = EtherDenomination.ETH,
+  }: {
+    rpcUrl: string;
+    chainId: string;
+    nickname: string;
+    rpcPrefs: RPCDefinition['rpcPrefs'];
+    ticker: string;
+  },
+  {
+    setActive,
+    source,
+  }: {
+    setActive: boolean;
+    source?: string;
+  },
 ): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
   return async (dispatch: MetaMaskReduxDispatch) => {
     log.debug(
-      `background.updateAndSetCustomRpc: ${newRpcUrl} ${chainId} ${
-        ticker ?? EtherDenomination.ETH
-      } ${nickname}`,
+      `background.upsertNetworkConfiguration: ${rpcUrl} ${chainId} ${ticker} ${nickname}`,
     );
 
     try {
-      await submitRequestToBackground('updateAndSetCustomRpc', [
-        newRpcUrl,
-        chainId,
-        ticker ?? EtherDenomination.ETH,
-        nickname || newRpcUrl,
-        rpcPrefs,
-      ]);
+      const networkConfigurationId: string = await submitRequestToBackground(
+        'upsertNetworkConfiguration',
+        [
+          { rpcUrl, chainId, ticker, nickname: nickname || rpcUrl, rpcPrefs },
+          { setActive, source },
+        ],
+      );
+      dispatch(setNewNetworkAdded({ networkConfigurationId, nickname }));
```

Comment:
Hmm. Do we know here that this network is new?

Reply:
Previously there was an `addCustomNetwork` function that was used only in cases where a new network was added. We could restore that, and have that action call `upsertNetworkConfiguration` internally. We can then call `setNewNetworkAdded` from there with confidence that it's only called for new networks.

Reply:
removed here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/ccfc91787ab01012269c2d2bd97fa9b3c0de5ace

And I've simply added the `setNewNetworkAdded` call as a followup to the `upsertNetworkConfiguration` call here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/ccfc91787ab01012269c2d2bd97fa9b3c0de5ace#diff-a225980bfdbe3e5ae6b7683796015494d01038cdc35d9245cc2d61436d5116c1R398 as we do in the network-form here: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/settings/networks-tab/networks-form/networks-form.js#L543